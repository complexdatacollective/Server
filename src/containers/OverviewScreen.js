import React from 'react';

const OverviewScreen = () => (
  <div className="screen">
    <div className="screen__heading">
      <h1 className="screen__heading-title">Network Canvas</h1>
      <h2 className="screen__heading-subtitle">Overview</h2>
    </div>
    <div className="screen__main">
      Various stats and info go here.
    </div>
  </div>
);

export default OverviewScreen;
