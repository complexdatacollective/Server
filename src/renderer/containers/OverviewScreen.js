import React from 'react';
import { OverviewDashboard } from '.';
import { Header } from '../components';

const OverviewScreen = () => (
  <div className="screen">
    <Header />
    <main className="screen__main">
      <OverviewDashboard />
    </main>
  </div>
);

export default OverviewScreen;
