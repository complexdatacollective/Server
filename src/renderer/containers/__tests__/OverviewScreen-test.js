/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import { UnconnectedOverviewScreen as OverviewScreen } from '../OverviewScreen';

describe('<OverviewScreen />', () => {
  const loadDevices = jest.fn();

  it('renders nothing while loading', () => {
    const subject = shallow(<OverviewScreen loadDevices={loadDevices} />);
    expect(subject.find('Instructions')).toHaveLength(0);
  });

  it('renders startup instructions when empty data loaded', () => {
    const props = { loadDevices, devices: [] };
    const subject = shallow(<OverviewScreen {...props} />);
    expect(subject.find('Instructions')).toHaveLength(1);
  });

  it('loads devices on startup', () => {
    shallow(<OverviewScreen loadDevices={loadDevices} />);
    expect(loadDevices).toHaveBeenCalled();
  });

  it('redirects to a protocol after startup', () => {
    const mockProtocol = { id: 'abc123', name: '1', createdAt: new Date(), updatedAt: new Date() };
    const subject = shallow((
      <OverviewScreen loadDevices={loadDevices} protocols={[mockProtocol]} />
    ));
    expect(subject.find('Instructions')).toHaveLength(0);
    expect(subject.find('Redirect')).toHaveLength(1);
    expect(subject.find('Redirect').prop('to')).toMatch(mockProtocol.id);
  });
});
