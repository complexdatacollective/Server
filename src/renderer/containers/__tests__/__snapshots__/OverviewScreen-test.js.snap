// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OverviewScreen /> should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OverviewScreen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="screen__heading"
>
          <h1
                    className="screen__heading-title"
          >
                    Network Canvas
          </h1>
          <h2
                    className="screen__heading-subtitle"
          >
                    Overview
          </h2>
</div>,
        <div
          className="screen__main"
>
          Various stats and info go here.
          <ServerPanel />
          <OverviewDashboard />
</div>,
      ],
      "className": "screen",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="screen__heading-title"
>
              Network Canvas
</h1>,
            <h2
              className="screen__heading-subtitle"
>
              Overview
</h2>,
          ],
          "className": "screen__heading",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Network Canvas",
              "className": "screen__heading-title",
            },
            "ref": null,
            "rendered": "Network Canvas",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Overview",
              "className": "screen__heading-subtitle",
            },
            "ref": null,
            "rendered": "Overview",
            "type": "h2",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Various stats and info go here.",
            <ServerPanel />,
            <OverviewDashboard />,
          ],
          "className": "screen__main",
        },
        "ref": null,
        "rendered": Array [
          "Various stats and info go here.",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="screen__heading"
>
            <h1
                        className="screen__heading-title"
            >
                        Network Canvas
            </h1>
            <h2
                        className="screen__heading-subtitle"
            >
                        Overview
            </h2>
</div>,
          <div
            className="screen__main"
>
            Various stats and info go here.
            <ServerPanel />
            <OverviewDashboard />
</div>,
        ],
        "className": "screen",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="screen__heading-title"
>
                Network Canvas
</h1>,
              <h2
                className="screen__heading-subtitle"
>
                Overview
</h2>,
            ],
            "className": "screen__heading",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Network Canvas",
                "className": "screen__heading-title",
              },
              "ref": null,
              "rendered": "Network Canvas",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Overview",
                "className": "screen__heading-subtitle",
              },
              "ref": null,
              "rendered": "Overview",
              "type": "h2",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Various stats and info go here.",
              <ServerPanel />,
              <OverviewDashboard />,
            ],
            "className": "screen__main",
          },
          "ref": null,
          "rendered": Array [
            "Various stats and info go here.",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
